docs/api/new-frameworks.md

```diff
@@ -0,0 +1,167 @@
+
+export default {
+  packageJson,
+  framework: ‘vue’,
+  frameworkPresets: [require.resolve('./framework-preset-vue.js')],
+};
+```
+<div style="background-color:#F8FAFC">
+TODO: VET REACT, VUE,WEB COMPONENTS links below
+</div>
+
+The value of the `framework` option (in this case ‘vue’) is something that gets passed to addons and allows them to do special case things for your framework.
+
+The real meat of this file is the framework presets, and these are standard [Storybook presets](../presets/introduction) -- you can look at framework packages in the Storybook monorepo (e.g. [react], [vue], [web-components]) to see examples of framework-specific customizations.
+
+
+### Configuring the client
+
+To configure the client, you must provide a framework specific render function. Before diving into the details, it’s important to understand how user-written stories relate to what is finally rendered on the screen.
+
+#### Renderable objects
+
+Storybook stories are ES6 functions that return a “renderable object.”
+
+Consider the following React story:
+
+```js
+// Button.story.js
+import { Button } from './Button';
+export default { 
+    title: 'Button', 
+    component: Button 
+}
+
+export const Sample = () => (
+  <Button label='hello button' />
+);
+```
+
+In this case, the renderable object is the React element, `<Button .../>`.
+
+In most other frameworks, the renderable object is actually a plain old javascript object. 
+
+Consider the following hypothetical example:
+
+```js
+// Button.story.js
+
+import { Button } from './Button';
+export default { 
+    title: 'Button', 
+    component: Button 
+}
+export const Sample = () => ({
+  template: '<button label=:label />',
+  data: {
+      label: 'hello button'
+    }
+});
+```
+
+The design of this “renderable object” is framework-specific, and should ideally match the idioms of that framework.
+
+#### Render function
+
+The frameworks render function is the thing responsible for converting the renderable object into DOM nodes. This is typically of the form:
+
+```js
+const rootElement = document.getElementById('root');
+
+export default function renderMain({
+  storyFn,
+}: RenderMainArgs) {
+  const storyObj = storyFn();
+  const html = fn(storyObj);
+  rootElement.innerHTML = html;
+}
+```
+
+#### Package structure
+
+On the client side, the key file is [`src/client/preview.js`](../configure/overview#configure-story-rendering):
+
+```js
+import { start } from '@storybook/core/client';
+import './globals';
+import render from './render';
+const api = start(render);
+
+// the boilerplate code
+```
+
+The globals file typically sets up a single global variable that client-side code (such as addon-provided decorators) can refer to if needed to understand which framework its running in:
+
+```js
+import { window } from 'global';
+window.STORYBOOK_ENV = ‘vue’;
+```
+
+<div style="background-color:#F8FAFC">
+TODO: need location on the render function for the frameworks
```

Comment:
- https://github.com/storybookjs/storybook/blob/next/app/react/src/client/preview/render.tsx
- https://github.com/storybookjs/storybook/blob/next/app/angular/src/client/preview/render.ts
- https://github.com/storybookjs/storybook/blob/next/app/vue/src/client/preview/render.ts

et al.

Reply:
Thanks for the links. They were added. Marking this as resolved